elasticsearch_test.go

```diff
@@ -47,105 +45,78 @@ func (e *UnitTestEnv) TestElasticsearch_Type(t *testing.T) {
 	}
 }
 
-func (e *UnitTestEnv) TestElasticsearch_Init(t *testing.T) {
-	config := map[string]interface{}{
-		"username": e.Username,
-		"password": e.Password,
-		"url":      e.URL,
-	}
-	configToStore, err := e.Elasticsearch.Init(context.Background(), config, true)
-	if err != nil {
-		t.Fatal(err)
+func (e *UnitTestEnv) TestElasticsearch_Initialize(t *testing.T) {
+	req := newdbplugin.InitializeRequest{
+		Config: map[string]interface{}{
+			"username": e.Username,
+			"password": e.Password,
+			"url":      e.URL,
+		},
+		VerifyConnection: true,
 	}
-	if len(configToStore) != len(config) {
-		t.Fatalf("expected %s, received %s", config, configToStore)
+	resp := dbtesting.AssertInitialize(t, e.Elasticsearch, req)
+	if len(resp.Config) != len(req.Config) {
```

Comment:
Since this isn't going through a translation layer (like gRPC), the `req.Config` could be manipulated in place, making these assertions useless. I recommend making an expected response object with a copy of the request config then comparing the two.

Reply:
Ah, ok, done in 507cc5319bdb4adbd706ad866cd591d087de9dfc